airbyte-integrations/bases/base-singer/base_singer/singer_helpers.py

```diff
@@ -73,7 +90,22 @@ def singer_catalog_to_airbyte_catalog(singer_catalog: Dict[str, any]) -> Airbyte
         for stream in singer_catalog.get("streams"):
             name = stream.get("stream")
             schema = stream.get("schema")
-            airbyte_streams += [AirbyteStream(name=name, json_schema=schema)]
+            airbyte_stream = AirbyteStream(name=name, json_schema=schema)
+
+            metadatas = stream.get("metadata")
+            stream_metadata_container = get_stream_level_metadata(metadatas) if metadatas else None
+            stream_metadata = stream_metadata_container.get("metadata")
+            if stream_metadata:
+                # TODO unclear from the singer spec what behavior should be if there are no valid replication keys, but forced-replication-method is INCREMENTAL.
+                #  For now requiring replication keys for a stream to be considered incremental.
+                replication_keys = stream_metadata.get("valid-replication-keys", [])
+                if len(replication_keys) > 0:
+                    airbyte_stream.source_defined_cursor = True
+                    airbyte_stream.supported_sync_modes = [SyncMode.full_refresh, SyncMode.incremental]
+                    # TODO if there are multiple replication keys, allow configuring which one is used. For now we deterministically take the first
```

Comment:
how common is this? will we likely get through most of our api integrations without having to do this? i like the approach. let's only bother doing it once we encounter a tap that forces us to.

Reply:
haven't found a source that does multiple keys yet